receiver/statsdreceiver/protocol/statsd_parser.go

```diff
@@ -258,6 +260,8 @@ func (p *StatsDParser) observerCategoryFor(t MetricType) ObserverCategory {
 		return p.histogramEvents
 	case TimingType:
 		return p.timerEvents
+	case CounterType:
+	case GaugeType:
```

Comment:
Same statement: this was intentional and I'm not sure what the linter is adding.